registrations/serializers.py

```diff
@@ -158,3 +163,37 @@ def save(self):
                 email
             )
         )
+
+class UserExternalTokenSerializer(serializers.ModelSerializer):
+    user_details = UserNameSerializer(source="user", read_only=True)
+    class Meta:
+        model = UserExternalToken
+        fields = 'id', 'user_details', 'token', 'expire_timestamp'
+
+    def validate_expire_timestamp(self, data):
+        if data < timezone.now():
+            raise serializers.ValidationError('Expire timestamp must be in the future.')
+        return data
+
+    def create(self, validated_data):
+        validated_data['user'] = self.context['request'].user
+
+        if not validated_data.get('expire_timestamp'):
+            validated_data['expire_timestamp'] = timezone.now() + timedelta(days=7)
+        
+        # Check if private and public key exists
+        if not(settings.JWT_PRIVATE_KEY and settings.JWT_PUBLIC_KEY):
+            raise serializers.ValidationError('Please Provide private and public key.')
+        
+        payload = {
+            'user': self.context['request'].user.id,
+            'exp': validated_data['expire_timestamp']
+        }
+
+        validated_data['token'] = jwt_encode_handler(payload)
```

Comment:
We don't store the token to the database, we only need to send it back when creating a new token.
NOTE: We store jit instead